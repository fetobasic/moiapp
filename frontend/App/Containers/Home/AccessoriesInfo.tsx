import React from 'react';
import { View, ScrollView, StyleSheet, Text } from 'react-native';
import { HeaderSimple as Header, Info } from 'App/Components';
import { ApplicationStyles, Colors, Fonts, Metrics } from 'App/Themes';
import renderElement from 'App/Services/RenderElement';
import WorldIcon from 'App/Images/Icons/world.svg';
import CombinerIcon from 'App/Images/Icons/combiner24.svg';
import TanksIcon from 'App/Images/Icons/tanks24.svg';
import { openLink } from 'App/Services/Linking';
import Links from 'App/Config/Links';

function AccessoriesInfo() {
  const colorDisabled = Colors.transparentWhite('0.87');

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header title="Accessories" />
      <ScrollView>
        <View style={styles.sectionInfo}>
          <View style={ApplicationStyles.row}>
            <WorldIcon />
            <Text style={styles.textTitle}>Goal Zero</Text>
          </View>
          <Text style={styles.textSubtitle}>
            Visit the website of Goal Zero to view and order more devices.{' '}
            <Text style={styles.textLink} onPress={() => openLink(Links.goalZeroHome)}>
              www.goalzero.com
            </Text>
          </Text>
        </View>
        <Info
          icon={renderElement(<CombinerIcon color={colorDisabled} />)}
          title="Combiner"
          description="Yeti inverter is paired with another Yeti inverter AND They're synchronized and ready to use in PARALLEL."
        />
        <Info
          info="LINK"
          title="Yeti Link"
          description="Recharge from the sun by connecting a compatible solar panel. Charge times are dependent on the size of the solar panel."
        />
        <Info
          icon={renderElement(<TanksIcon color={colorDisabled} />)}
          title="Tanks"
          description="Each Yeti Tank Lead Acid battery adds 1,200 Watt Hours of energy storage to your system. "
        />
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  sectionInfo: {
    marginTop: Metrics.marginSection,
    marginHorizontal: Metrics.baseMargin,
  },
  textTitle: {
    marginLeft: Metrics.halfMargin,
    ...Fonts.font.condensed.h3,
  },
  textSubtitle: {
    marginTop: Metrics.smallMargin,
    marginLeft: 48,
    ...Fonts.font.base.bodyOne,
    color: Colors.transparentWhite('0.87'),
  },
  textLink: {
    color: Colors.green,
  },
});

export default AccessoriesInfo;
