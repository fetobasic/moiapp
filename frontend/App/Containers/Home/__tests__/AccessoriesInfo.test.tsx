import React from 'react';
import { render } from '@testing-library/react-native';
import AccessoriesInfo from '../AccessoriesInfo';

describe('AccessoriesInfo', () => {
  test('renders screen', () => {
    const tree = render(<AccessoriesInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
