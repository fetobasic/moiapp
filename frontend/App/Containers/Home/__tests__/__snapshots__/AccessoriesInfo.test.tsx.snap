// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessoriesInfo renders screen 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Accessories
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "marginHorizontal": 24,
            "marginTop": 16,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <SvgMock />
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "RobotoCondensed-Medium",
                "fontSize": 20,
                "fontWeight": "500",
                "lineHeight": 32,
                "marginLeft": 12,
              }
            }
          >
            Goal Zero
          </Text>
        </View>
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 0.87)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginLeft": 48,
              "marginTop": 8,
            }
          }
        >
          Visit the website of Goal Zero to view and order more devices.
           
          <Text
            onPress={[Function]}
            style={
              {
                "color": "rgba(191, 210, 43, 1)",
              }
            }
          >
            www.goalzero.com
          </Text>
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock
                color="rgba(255, 255, 255, 0.87)"
              />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Combiner
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Yeti inverter is paired with another Yeti inverter AND They're synchronized and ready to use in PARALLEL.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgba(255, 255, 255, 0.6)",
                  "fontFamily": "RobotoCondensed-Medium",
                  "fontSize": 16,
                  "fontWeight": "700",
                  "lineHeight": 24,
                }
              }
            >
              LINK
            </Text>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Yeti Link
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Recharge from the sun by connecting a compatible solar panel. Charge times are dependent on the size of the solar panel.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock
                color="rgba(255, 255, 255, 0.87)"
              />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Tanks
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Each Yeti Tank Lead Acid battery adds 1,200 Watt Hours of energy storage to your system. 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
